File: 2683-neighboring-bitwise-xor/2683-neighboring-bitwise-xor.java
class Solution {
    public boolean doesValidArrayExist(int[] derived) {
        int x = 0;
        int y = 1;
        for(int i=0; i<derived.length-1; i++)
        {
            x ^= derived[i];
            y ^= derived[i];
        }
        if((x^0) == derived[derived.length-1]) return true;
        if((y^1) == derived[derived.length-1]) return true;
        return false;
    }
}

File: 2683-neighboring-bitwise-xor/2683-neighboring-bitwise-xor.py
class Solution:
    def doesValidArrayExist(self, derived: List[int]) -> bool:
        x = 0
        y = 1
        for i in range(len(derived)-1):
            x ^= derived[i]
            y ^= derived[i]
        
        if (x^0) == derived[-1]: return True
        if (y^1) == derived[-1]: return True
        return False